aws-android-sdk-auth-core/src/main/java/com/amazonaws/mobile/auth/core/IdentityManager.java

```diff
@@ -122,8 +121,7 @@ private void setUnderlyingProvider(final CognitoCachingCredentialsProvider under
     private final CountDownLatch startupAuthTimeoutLatch = new CountDownLatch(1);
 
     /** Keep track of the registered sign-in providers. */
-    private final List<Class<? extends SignInProvider>> signInProviderClasses
-        = new LinkedList<Class<? extends SignInProvider>>();
+    private final Set<Class<? extends SignInProvider>> signInProviderClasses = new HashSet<>();
```

Comment:
Why use `Set` to store `signInProviderClasses` instead of `List`?

Reply:
It removes the need to check for duplicates in `registerConfigSignInProviders()`. `getSignInProviderClasses()` returns a `Collection` anyways, so this is not a breaking change.